packages/flutter/lib/src/rendering/child_view.dart

```diff
@@ -20,7 +20,7 @@ import 'object.dart';
 mojom.ViewProxy _initViewProxy() {
   int viewHandle = ui.takeViewHandle();
   assert(() {
-    if (viewHandle == 0)
+    if (viewHandle == core.MojoHandle.INVALID)
       debugPrint('Child view are supported only when running in Mojo shell.');
```

Comment:
That makes a certain amount of sense.  This code is inside an assert(), which means it only executes in checked mode.  It's also inside a function that executes at most once per run of the app.  I guess when I wrote it, I was thinking it was a debugging aid, similar to dumpRenderTree.

Maybe the right thing to do is to show this message through the UI rather than through the console.  For example, in checked mode outside Mojo shell, we could replace the contents of the ChildView widget with this message.